### DISCUSSION TITLE
How to get all models from a specific relationship ?

### DISCUSSION THREAD

**Discussion by ArthurDanjou** [adonisjs/core#1991] [Help] [answered]
Hello, I will explain you my problem. I have two models, a Post model and a Tag model. Models are linked with a many to many relationship.

Here is my post model : https://hastebin.com/nogivovuri.typescript
Here is my tag model : https://hastebin.com/uretaxebid.typescript

When i get posts from a get http request with the tag in params, I dont know how to get only the posts which contains the tag in the pivot table. I have test many methods. 

Here is my controller : https://hastebin.com/igiparajat.typescript

In short, I want to get all posts with the tag model linked.

Thank's for help !

**Comment by thetutlage** [comment]
I believe, you should be using the `tag` instance to fetch the related posts. This is how it will work conceptually.

- `Tag` hasMany `Post`
- Then use the tag instance to fetch posts.
   ```ts
   tag.preload('posts')
   ```

**Reply by ArthurDanjou** [reply]
As you can see, my Post model already has the manyToMany relation with tags. Do I need to add the hasMany relation in the tag model ?

**Comment by ArthurDanjou** [comment, marked as answer]
I have finally found the solution ! 
```return Post.query()
        .preload('tags', (tag) => {
          tag.preload('label')
        })
        .whereHas('tags', (tag) => {
          tag.wherePivot('tag_id', queriedTag.id)
        })
        .preload('cover')
        .preload('title')
        .preload('description')
        .preload('content')
        .paginate(page, 5)```


Here is my new method
